.github/workflows/run-test-harness.yml

```diff
@@ -16,11 +16,19 @@ jobs:
                   fetch-depth: 0
             - name: Install protoc for protobuf compilation
               run: sudo apt-get install -y protobuf-compiler
-            - name: Run Yarn
-              run: yarn --immutable
+            - uses: pnpm/action-setup@v2
+              with:
+                  version: 8.6.12
+            - name: Use Node.js ${{ matrix.node-version }}
+              uses: actions/setup-node@v3
+              with:
+                  node-version: ${{ matrix.node-version }}
+                  cache: 'pnpm'
```

Comment:
Do you not have to reverse these two?

Reply:
no, you need pnpm installed in order to run setup node with the pnpm cache option